JsonMappingException on missing userIdentity in CloudTrailRecord

In HS-1155337333 a customer has problems because the `userIdentity` field is missing when deserializing the `CloudTrailRecord`. A `NullPointerException` occurs when constructing the `construcedMessage` [field](https://github.com/Graylog2/graylog-plugin-aws/blob/740d47c50badd0659a364d04d790d7465e2ea9af/src/main/java/org/graylog/aws/inputs/cloudtrail/json/CloudTrailRecord.java#L113). Here is the StackTrace:
```
2022-10-24T13:25:48.996-04:00 ERROR [CloudTrailSubscriber] Could not read CloudTrail log file for <cloudtrail-graylog>. Skipping.
com.fasterxml.jackson.databind.JsonMappingException: (was java.lang.NullPointerException) (through reference chain: [org.graylog.aws](http://org.graylog.aws/).inputs.cloudtrail.json.CloudTrailRecord["constructedMessage"])
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath([JsonMappingException.java:394](http://jsonmappingexception.java:394/)) ~[graylog.jar:?]
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath([JsonMappingException.java:353](http://jsonmappingexception.java:353/)) ~[graylog.jar:?]
        at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow([StdSerializer.java:316](http://stdserializer.java:316/)) ~[graylog.jar:?]
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields([BeanSerializerBase.java:727](http://beanserializerbase.java:727/)) ~[graylog.jar:?]
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize([BeanSerializer.java:155](http://beanserializer.java:155/)) ~[graylog.jar:?]
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize([DefaultSerializerProvider.java:480](http://defaultserializerprovider.java:480/)) ~[graylog.jar:?]
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue([DefaultSerializerProvider.java:319](http://defaultserializerprovider.java:319/)) ~[graylog.jar:?]
        at com.fasterxml.jackson.databind.ObjectMapper._configAndWriteValue([ObjectMapper.java:3906](http://objectmapper.java:3906/)) ~[graylog.jar:?]
        at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsBytes([ObjectMapper.java:3244](http://objectmapper.java:3244/)) ~[graylog.jar:?]
        at [org.graylog.aws.inputs.cloudtrail.CloudTrailSubscriber.run](http://org.graylog.aws.inputs.cloudtrail.cloudtrailsubscriber.run/)([CloudTrailSubscriber.java:157](http://cloudtrailsubscriber.java:157/)) [graylog-plugin-aws-4.3.8.jar:?]
Caused by: java.lang.NullPointerException
        at [org.graylog.aws](http://org.graylog.aws/).inputs.cloudtrail.json.CloudTrailRecord.getConstructedMessage([CloudTrailRecord.java:113](http://cloudtrailrecord.java:113/)) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor320.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke([DelegatingMethodAccessorImpl.java:43](http://delegatingmethodaccessorimpl.java:43/)) ~[?:?]
        at java.lang.reflect.Method.invoke([Method.java:566](http://method.java:566/)) ~[?:?]
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField([BeanPropertyWriter.java:688](http://beanpropertywriter.java:688/)) ~[graylog.jar:?]
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields([BeanSerializerBase.java:719](http://beanserializerbase.java:719/)) ~[graylog.jar:?]
        ... 6 more
```

--- a/changelog/unreleased/issue-667.toml
+++ b/changelog/unreleased/issue-667.toml
@@ -1,0 +1,4 @@
+type = "fixed"
+message = "Fix JSON deserialization errors in AWS CloudTrail input."
+
+pulls = ["677"]

--- a/changelog/unreleased/issue-667.toml
+++ b/changelog/unreleased/issue-667.toml
@@ -1,4 +1,5 @@
 type = "fixed"
 message = "Fix JSON deserialization errors in AWS CloudTrail input."
 
+issues = ["667"]
 pulls = ["677"]

--- a/src/main/java/org/graylog/aws/inputs/cloudtrail/json/CloudTrailRecord.java
+++ b/src/main/java/org/graylog/aws/inputs/cloudtrail/json/CloudTrailRecord.java
@@ -23,6 +23,7 @@ import com.graylog2.input.cloudtrail.json.CloudTrailResponseElements;
 import java.io.Serializable;
 import java.util.Arrays;
 import java.util.Map;
+import java.util.Optional;
 
 public class CloudTrailRecord implements Serializable {
     @JsonProperty("eventVersion")
@@ -59,7 +60,7 @@ public class CloudTrailRecord implements Serializable {
     @JsonProperty("additionalEventData")
     public Map<String, Object> additionalEventData;
 
-    //adding errorMessage 
+    //adding errorMessage
     @JsonProperty("errorMessage")
     public String errorMessage;
 
@@ -110,7 +111,8 @@ public class CloudTrailRecord implements Serializable {
     }
 
     public String getConstructedMessage() {
-        return eventSource + ":" + eventName + " in " + awsRegion + " by " + sourceIPAddress + " / " + userIdentity.userName;
+        return eventSource + ":" + eventName + " in " + awsRegion + " by " + sourceIPAddress + " / " +
+                Optional.ofNullable(userIdentity).map(i -> i.userName).orElse("<unknown user_name>");
     }
 
 }

--- a/src/main/java/org/graylog/aws/inputs/cloudtrail/notifications/CloudtrailSNSNotificationParser.java
+++ b/src/main/java/org/graylog/aws/inputs/cloudtrail/notifications/CloudtrailSNSNotificationParser.java
@@ -29,8 +29,11 @@ import java.util.Collections;
 import java.util.List;
 
 public class CloudtrailSNSNotificationParser {
-    private final ObjectMapper objectMapper;
+
+    private static final String CLOUD_TRAIL_VALIDATION_MESSAGE = "CloudTrail validation message.";
     private static final Logger LOG = LoggerFactory.getLogger(CloudtrailSNSNotificationParser.class);
+
+    private final ObjectMapper objectMapper;
 
     public CloudtrailSNSNotificationParser(ObjectMapper objectMapper) {
         this.objectMapper = objectMapper;
@@ -49,6 +52,10 @@ public class CloudtrailSNSNotificationParser {
             }
 
             LOG.debug("Reading message envelope {}.", envelope.message);
+            if (envelope.message.contains(CLOUD_TRAIL_VALIDATION_MESSAGE)) {
+                return Collections.emptyList();
+            }
+
             final CloudtrailWriteNotification notification = objectMapper.readValue(envelope.message, CloudtrailWriteNotification.class);
 
             final List<String> s3ObjectKeys = notification.s3ObjectKey;

